src/FirebirdSql.Data.FirebirdClient/Client/Managed/Version10/GdsBlob.cs

```diff
@@ -216,7 +300,142 @@ protected override async ValueTask GetSegmentAsync(Stream stream, CancellationTo
 		}
 	}
 
-	protected override void PutSegment(byte[] buffer)
+	internal override byte[] GetSegment()
+	{
+		var requested = SegmentSize;
+
+#if !(NET48 || NETSTANDARD2_0)
+		var tmp = ArrayPool<byte>.Shared.Rent(requested * 2);
```

Comment:
I would remove the `ArrayPool` from this PR. It's not used anywhere, yet. It might be bigger sweep later across the codebase.

Reply:
addressed